agent-core/src/main/java/com/newrelic/agent/android/agentdata/AgentDataReporter.java

```diff
@@ -69,8 +64,19 @@ public static boolean reportAgentData(byte[] bytes) {
         if (isInitialized()) {
             if (reportExceptions) {
                 Payload payload = new Payload(bytes);
-                instance.get().storeAndReportAgentData(payload);
-                reported = true;
+                if (FeatureFlag.featureEnabled(FeatureFlag.OfflineStorage)) {
```

Comment:
You really done when y need to understand the Payload controller. Your implementation simply duplicates what is already being done when `instance.get().storeAndReportAgentData(payload)` is called (hence the _store_ parth of the method name). Is there a reason to store the data elsewhere?